xapian-core/cluster/cluster.cc

```diff
@@ -296,10 +296,10 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	string term = *it;
 	double termfreq = tlg.get_termfreq(term);
 
-	/** If term is not a stemmed term or the term indexes only one
-	 *  one document, do not compute TF-IDF scores for them
-	 */
-	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
+	/// If term frequency of the term is lesser than or equal
+	/// to 1 or wdf is lesser than 1, we do not consider this
+	/// term for document vector calculations
+	if (wdf < 1 || termfreq <= 1)
 	    continue;
```

Comment:
Oh yes I checked again and the the terms were stemmed without the Z prefix. I wasn't aware about this.

> You can create a test database using TermGenerator in a more common mode - just use a "generated" database where you pass a function to do the indexing. The harness will then take care of caching it, only calling that function if it doesn't have a suitable cached database to use.

Great. Would it be better to add this in along with stopword removal? I guess its better having other parts of the clusterer in place before we start dealing with stopword removal and stemming.